docs/vendor/gcp/setup.sh

```diff
@@ -0,0 +1,155 @@
+      --repository-format=docker --location="us"
+fi
+
+# GCP settings
+# Ensure IAM is enabled
+echo "Enabling IAM services."
+gcloud services enable iam.googleapis.com
+
+# Create a Kubernetes Service account
+echo "Create a kubernetes service account ${KSA_NAME}"
+if kubectl get serviceAccount "${KSA_NAME}"; then
+  echo "KSA ${KSA_NAME} already exists"
+else
+  kubectl create serviceaccount "${KSA_NAME}" --namespace "${NAMESPACE}"
+  echo "Successfully created Kubernetes service account ${KSA_NAME} in ${NAMESPACE}."
+fi
+
+# Create a GCloud Service account
+echo "Create a Google service account ${GSA_NAME}"
+if gcloud iam service-accounts list | grep "${GSA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com"; then
+  echo "GSA ${GSA_NAME} already exists"
+else
+  gcloud iam service-accounts create "${GSA_NAME}" \
+    --description="Tekton Build-time Service Account" \
+    --display-name="Tekton Builder"
+  echo "Successfully created a Google service account ${GSA_NAME}"
+fi
+
+echo "Provide aritifactregistry.writer access to service account ${GSA_NAME}"
+# Provide necessary permissions to the KSA Service account.
+gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+  --member "serviceAccount:${GSA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com" \
+  --role "roles/artifactregistry.writer"
+
+echo "Provide cloudkms.cryptoOperator access to service account ${GSA_NAME}"
+gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+  --member "serviceAccount:${GSA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com" \
+  --role "roles/cloudkms.cryptoOperator"
+
+echo "Provide cloudkms.viewer access to service account ${GSA_NAME}"
+gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+  --member "serviceAccount:${GSA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com" \
+  --role "roles/cloudkms.viewer"
+
+echo "Provide containeranalysis.admin access to service account ${GSA_NAME}"
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+  --member "serviceAccount:$GSA_NAME@$PROJECT_ID.iam.gserviceaccount.com" \
+  --role "roles/containeranalysis.admin"
+
+echo "Adding workloadIndentityUser access to Kubernetes Service Account ${KSA_NAME} for Google Service account ${GSA_NAME}"
+gcloud iam service-accounts add-iam-policy-binding $GSA_NAME@$PROJECT_ID.iam.gserviceaccount.com \
+  --role "roles/iam.workloadIdentityUser" \
+  --member "serviceAccount:$PROJECT_ID.svc.id.goog[$NAMESPACE/$KSA_NAME]"
+
+echo "Annotate the Kubernetes Service Account ${KSA_NAME} to point to the Google Service Account ${GSA_NAME} to use."
+kubectl annotate serviceaccount $KSA_NAME \
+  --overwrite \
+  --namespace $NAMESPACE \
+  iam.gke.io/gcp-service-account=$GSA_NAME@$PROJECT_ID.iam.gserviceaccount.com
+
+# install pipelines
+kubectl apply --filename https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
+# install chains
+kubectl apply -f https://storage.googleapis.com/tekton-releases-nightly/chains/previous/v20221215-6d3449f5ea/release.yaml
```

Comment:
If you're planning to use this for a blog post, don't use latest/nightly so that this can be repeatable.